### DISCUSSION TITLE
[New Feature] Cancel all queue.

### DISCUSSION THREAD

**Discussion by Lylaktech** [Vinifera-Developers/Vinifera#1468] [Feature Requests]
### Description:
Pressing a hotkey and right clicking a unit twice should cancel all units of that type in the queue.

**Comment by CCHyper** [comment]
Great QoL feature and I have investigated this in Red Alert 2 already. There is quite a lot involved and a new network event was added for just this purpose, so I'll have to reimplement quite a few functions for this to work.

**Comment by JoyfulShush** [comment]
This is now partially possible thanks to the Shift Queue feature - shift right clicking on a queue removes all unit queues of that type. From Vinifera's docs:

`You can hold CONTROL while dequeueing to dequeue 5 units at a time, or SHIFT to dequeue all units of that type.`

That said, extending it so a button such as ALT would dequeue everything for that queue might be an interesting addition.
